rpc/handlers.go

```diff
@@ -1528,10 +1588,20 @@ func (h *Handler) simulateTransactions(id BlockID, transactions []BroadcastedTra
 			}
 		}
 
+		var dataGasPrice *felt.Felt
+		switch feeUnit {
+		case FRI:
+			dataGasPrice = header.L1DataGasPrice.PriceInFri
+		case WEI:
+			dataGasPrice = header.L1DataGasPrice.PriceInWei
+		}
+
 		estimate := FeeEstimate{
-			GasConsumed: new(felt.Felt).Div(overallFee, gasPrice),
-			GasPrice:    gasPrice,
-			OverallFee:  overallFee,
+			GasConsumed:     new(felt.Felt).Div(overallFee, gasPrice),
+			GasPrice:        gasPrice,
+			DataGasConsumed: new(felt.Felt).Div(overallFee, dataGasPrice), // please double check that
```

Comment:
🙏

Reply:
overall_fee will now be gas_consumed × gas_price + data_gas_consumed × data_gas_price

Reply:
Recent changes look good